thirdeye-ui/src/app/components/admin-page/recent-failures/recent-failures.component.tsx

```diff
@@ -12,31 +12,110 @@
  * See the License for the specific language governing permissions and limitations under
  * the License.
  */
-import { Box, Grid, Table, Typography } from "@material-ui/core";
+import {
+    Box,
+    Grid,
+    Table,
+    Toolbar,
+    Typography,
+    useMediaQuery,
+    useTheme,
+} from "@material-ui/core";
 import TableBody from "@material-ui/core/TableBody";
 import TableCell from "@material-ui/core/TableCell";
 import TableHead from "@material-ui/core/TableHead";
 import TableRow from "@material-ui/core/TableRow";
+import classNames from "classnames";
 import { sortBy } from "lodash";
 import React, { FunctionComponent, useEffect, useMemo } from "react";
 import { useTranslation } from "react-i18next";
+import { useSearchParams } from "react-router-dom";
 import { PageContentsCardV1, SkeletonV1 } from "../../../platform/components";
+import { useDataGridV1Styles } from "../../../platform/components/data-grid-v1/data-grid-v1/data-grid-v1.styles";
 import { ActionStatus } from "../../../rest/actions.interfaces";
 import { TaskStatus, TaskType } from "../../../rest/dto/taks.interface";
 import { useGetTasks } from "../../../rest/tasks/tasks.actions";
+import { getTimeRangeDuration } from "../../../utils/time-range/time-range.util";
+import { WEEK_IN_MILLISECONDS } from "../../../utils/time/time.util";
 import { NoDataIndicator } from "../../no-data-indicator/no-data-indicator.component";
 import { LoadingErrorStateSwitch } from "../../page-states/loading-error-state-switch/loading-error-state-switch.component";
+import { useTimeRange } from "../../time-range/time-range-provider/time-range-provider.component";
+import {
+    TimeRange,
+    TimeRangeDuration,
+    TimeRangeQueryStringKey,
+} from "../../time-range/time-range-provider/time-range-provider.interfaces";
+import { TimeRangeSelector } from "../../time-range/time-range-selector/time-range-selector/time-range-selector.component";
 import { TaskRow } from "./task-row/task-row.component";
 
 export const RecentFailures: FunctionComponent = () => {
+    const theme = useTheme();
+    const dataGridV1Classes = useDataGridV1Styles();
     const { t } = useTranslation();
     const { tasks, getTasks, status } = useGetTasks();
+    const screenWidthSmUp = useMediaQuery(theme.breakpoints.up("sm"));
+    const [searchParams, setSearchParams] = useSearchParams();
 
-    useEffect(() => {
+    const [startTime, endTime] = useMemo(
+        () => [
+            Number(searchParams.get(TimeRangeQueryStringKey.START_TIME)),
+            Number(searchParams.get(TimeRangeQueryStringKey.END_TIME)),
+        ],
+        [searchParams]
+    );
+
+    const {
+        timeRangeDuration,
+        recentCustomTimeRangeDurations,
+        setTimeRangeDuration,
+    } = useTimeRange();
```

Comment:
Don't use `useTimeRange `. I've been trying to move off this and strictly use whats in the search params.

Reply:
I've removed the TE-1002 (date filter in admin) code to [#774](https://github.com/startreedata/thirdeye/pull/774), will address this comment there